pkg/controller/operators/catalog/operator.go

```diff
@@ -936,24 +942,52 @@ func (o *Operator) ensureSubscriptionCSVState(logger *logrus.Entry, sub *v1alpha
 	return updatedSub, true, nil
 }
 
-func (o *Operator) updateSubscriptionStatus(namespace string, subs []*v1alpha1.Subscription, installPlanRef *corev1.ObjectReference) error {
-	// TODO: parallel, sync waitgroup
-	var err error
+func (o *Operator) updateSubscriptionStatus(namespace string, gen int, subs []*v1alpha1.Subscription, installPlanRef *corev1.ObjectReference) error {
+	var (
+		errs        []error
+		mu          sync.Mutex
+		wg          sync.WaitGroup
+		getOpts     = metav1.GetOptions{}
+		lastUpdated = o.now()
+	)
 	for _, sub := range subs {
-		sub.Status.LastUpdated = o.now()
+		sub.Status.LastUpdated = lastUpdated
 		if installPlanRef != nil {
 			sub.Status.InstallPlanRef = installPlanRef
 			sub.Status.Install = v1alpha1.NewInstallPlanReference(installPlanRef)
 			sub.Status.State = v1alpha1.SubscriptionStateUpgradePending
+			sub.Status.InstallPlanGeneration = gen
 		}
-		if _, subErr := o.client.OperatorsV1alpha1().Subscriptions(namespace).UpdateStatus(sub); subErr != nil {
-			err = subErr
-		}
+
+		wg.Add(1)
+		go func(s v1alpha1.Subscription) {
+			defer wg.Done()
+
+			update := func() error {
+				// Update the status of the latest revision
+				latest, err := o.client.OperatorsV1alpha1().Subscriptions(s.GetNamespace()).Get(s.GetName(), getOpts)
```

Comment:
Are you seeing flakes in the e2e tests here?

We may want to use the ‘retry’ package here. Can’t the subscription status update loop conflict with this? 

Another option would be to determine what the current install plan generation is during the subscription update loop.